design/apidsl/media_type.go

```diff
@@ -376,9 +376,17 @@ func Link(name string, view ...string) {
 // CollectionOf creates a collection media type from its element media type. A collection media
```

Comment:
Please remove the first 3 lines:
```go
// CollectionOf can be used in: Wherever a MediaType can be used..
 //
 // e.g. Attribute("foo", CollectionOf(Bar))
```
(I know you didn't add them but they don't add much value and where `MediaType` can be used is already explained below). @deltaskelta fyi - we probably should remove them from the other places that say "wherever...".

Also we should probably change the first line to something like:
```go
// CollectionOf creates a collection media type from its element media type and an optional identifier.